stream_test.go

```diff
@@ -19,17 +23,16 @@ const (
 )
 
```

Comment:
There are some new tests here, but I've also rewritten the existing tests so that they no longer use an `eventsource.Server` as the other end of the end-to-end test, but instead use a much simpler HTTP handler (with support from the `go-test-helpers` types). This makes the tests here more self-contained and ensures that if Server's behavior changes in some way it doesn't produce false results in the Stream tests.